### PR TITLE
WELD-2386 allow to define bean discovery mode for synthetic archives …

### PR DESCRIPTION
**Pull request by manovotn**
…in SE environment.

### CODE REVIEW

File: environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java [hunk 1]
```diff
@@ -662,6 +664,17 @@ public Weld addServices(Service... services) {
     }
 
     /**
+     * Sets the bean discovery mode for synthetic bean archive. Default mode is ALL.
+     * @param mode Bean discovery mode as string, possible values are ALL, ANNOTATED, NONE
+     *
+     * @return  self
+     */
+    public Weld setSyntheticArchiveBeanDiscoveryMode(String mode) {
```
Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java]: I think `setBeanDiscoveryMode()` should be ok. We don't use `syntheticArchive` in other methods neither.

Comment (manovotn) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java]: Hmm, can do I just wanted to make it clear that this makes sense only once you `disableDiscovery()`

Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java]: Also `BeanDiscoveryMode` is part of SPI. So we could probably use the enum as the param.

Comment (tremes) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java]: +1 to enum

**Comment by tremes** [comment]
It's just minor, but I would add new SE tests to `environments/se/tests`

**Comment by manovotn** [comment]
@tremes I don' understand, that's where the tests are, isn't it?

**Comment by tremes** [comment]
Ah sorry you're right. I thought it was in `environments/se/core/test/`.

**Comment by manovotn** [comment]
Since weekend are long...a note to myself:
* Martin suggested a filter instead of mode as parameter - we could expand on this.
* I removed all notions of mode `NONE` as an option because it makes no sense in SE.
** Therefore I added an exception/test for IAE when you use this mode

File: environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java [hunk 2]
```diff
@@ -662,6 +665,21 @@ public Weld addServices(Service... services) {
     }
 
     /**
+     * Sets the bean discovery mode for synthetic bean archive. Default mode is ALL.
+     * @param mode Bean discovery mode as string, possible values are ALL, ANNOTATED
```
Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java]: It's not a string anymore...

File: environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java [hunk 3]
```diff
@@ -662,6 +665,21 @@ public Weld addServices(Service... services) {
...
+     *
+     * @return  self
+     */
```
Comment (mkouba) [inline environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java]: Add `@throws IllegalArgumentException If...`

**Comment by mkouba** [comment]
The idea of a filter is probably an overkill.

**Pull request merged**